bluephos/tasks/filter_pipeline.py

```diff
@@ -0,0 +1,69 @@
+import pandas as pd
+from dplutils.pipeline import PipelineTask
+from bluephos.modules.filter import FilterTask
+
+
+def apply_filter(df: pd.DataFrame, column_name, threshold: float, filter_in: bool) -> pd.DataFrame:
+    filter_task = FilterTask(column_name=column_name, threshold=threshold, filter_in=filter_in)
+    return filter_task.task(df)
+
+
+# Helper function to create both filter_in and filter_out tasks
+def create_filter_task(task_name: str, column_name: str, threshold_key: str, filter_in: bool) -> PipelineTask:
+    def apply_filter(df: pd.DataFrame, threshold: float) -> pd.DataFrame:
+        filter_task = FilterTask(column_name=column_name, threshold=threshold, filter_in=filter_in)
+        return filter_task.task(df)
+
+    return PipelineTask(
+        task_name,
+        apply_filter,
+        context_kwargs={
+            "threshold": threshold_key,  # Get threshold from the context
+        },
+    )
```

Comment:
I think this works fine, but looks like it might be a bit more machinery than needed. Could something like:

```py
def filter(df: DataFrame, column: str, threshold: float, filter_in: bool = True) -> DataFrame:
    filtered = df[column] < threshold
    if filtered_in:
        return df[filtered]
    return df[~filtered]
```

For the filter task, and then each task defined like:

```py
FilterNNInTask = PipelineTask(
  'filter_nn_in', filter, kwargs={'column': 'z'}, context_kwargs={'threshold': 't_nn'})

FilterNNOutTask = PipelineTask(
  'filter_nn_out', filter, kwargs={'column': 'z', 'filter_in': False}, context_kwargs={'threshold': 't_nn'})

...
```

The task definition looks similar amount of code, and quite simpler for the filter.

Reply:
Thanks! Simplified the codes.